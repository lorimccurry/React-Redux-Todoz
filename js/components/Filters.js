import React from 'react'
import { connect } from 'react-redux'
import { setActiveFilter } from '../actions/actionCreators'

class Filters extends React.Component {
  constructor () {
    super()

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (filter) {
    this.props.dispatchActiveFilter(filter)
  }

  render () {
    return (
      <div>
        <button
          className='filter-all'
          onClick={() => this.handleClick('all')}
        >
          All
        </button>
        <button
          className='filter-completed'
          onClick={() => this.handleClick('completed')}
        >
          Completed
        </button>
        <button
          className='filter-uncompleted'
          onClick={() => this.handleClick('uncompleted')}
        >
          Uncompleted
        </button>
      </div>
    )
  }
}

const { func } = React.PropTypes
Filters.propTypes = {
  dispatchActiveFilter: func
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatchActiveFilter: (filter) => {
      dispatch(setActiveFilter(filter))
    }
  }
}

export const Unwrapped = Filters

export default connect(null, mapDispatchToProps)(Filters)
