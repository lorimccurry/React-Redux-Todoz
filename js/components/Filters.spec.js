import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { spy } from 'sinon'
import Filters from './Filters' 
import { Unwrapped as UnwrappedFilters } from './Filters'

describe('Filters', () => {
  it('Filters snapsnot test', () => {
    const component = shallow(<UnwrappedFilters />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('renders All Button, Completed Button, Uncompleted Button', () => {
    const component = shallow(<UnwrappedFilters />) 
    expect (component.find('.filter-all').length).toEqual(1)
    expect (component.find('.filter-completed').length).toEqual(1)
    expect (component.find('.filter-uncompleted').length).toEqual(1)
  }) 

  it('should call dispatchFilterAll with all arg when All clicked', () => {
    const filterSpy = spy()
    const component = mount(<UnwrappedFilters dispatchActiveFilter={filterSpy} />) 
    const checkbox = component.find('.filter-all')
    checkbox.simulate('click')

    expect(filterSpy.calledOnce).toEqual(true)
    expect(filterSpy.calledWith('all')).toEqual(true)
  })

  it('should call dispatchFilterAll with completed arg when Completed clicked', () => {
    const filterSpy = spy()
    const component = mount(<UnwrappedFilters dispatchActiveFilter={filterSpy} />) 
    const checkbox = component.find('.filter-completed')
    checkbox.simulate('click')

    expect(filterSpy.calledOnce).toEqual(true)
    expect(filterSpy.calledWith('completed')).toEqual(true)
  })

  it('should call dispatchFilterAll with uncompleted arg when Uncompleted clicked', () => {
    const filterSpy = spy()
    const component = mount(<UnwrappedFilters dispatchActiveFilter={filterSpy} />) 
    const checkbox = component.find('.filter-uncompleted')
    checkbox.simulate('click')

    expect(filterSpy.calledOnce).toEqual(true)
    expect(filterSpy.calledWith('uncompleted')).toEqual(true)
  })
})
