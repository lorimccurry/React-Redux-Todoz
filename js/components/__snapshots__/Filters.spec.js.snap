// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Filters snapsnot test 1`] = `
<div>
  <button
    className="filter-all"
    onClick={[Function]}
  >
    All
  </button>
  <button
    className="filter-completed"
    onClick={[Function]}
  >
    Completed
  </button>
  <button
    className="filter-uncompleted"
    onClick={[Function]}
  >
    Uncompleted
  </button>
</div>
`;
